google-cloud-firestore/lib/google/cloud/firestore.rb

```diff
@@ -21,6 +21,381 @@ module Cloud
+    # none of them are applied. There are two types of atomic operations in
+    # Cloud Firestore: A transaction is a set of read and write operations on
+    # one or more documents, while a batched write is a set of only write
+    # operations on one or more documents. (For more information, see
+    # [Transactions and Batched Writes](https://cloud.google.com/firestore/docs/manage-data/transactions).
+    #
+    # ### Using transactions
+    #
+    # A transaction consists of any number of read operations followed by any
+    # number of write operations. (Read operations must always come before write
+    # operations.) In the case of a concurrent update by another client, Cloud
+    # Firestore runs the entire transaction again. Therefore, transaction blocks
+    # should be idempotent and should not not directly modify application state.
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # city = firestore.col("cities").doc("SF")
+    # city.set({ name: "San Francisco",
+    #            state: "CA",
+    #            country: "USA",
+    #            capital: false,
+    #            population: 860000 })
+    #
+    # firestore.transaction do |tx|
+    #   new_population = tx.get(city).data[:population] + 1
+    #   tx.update(city, { population: new_population })
+    # end
+    # ```
+    #
+    # ### Using batched writes
+    #
+    # If you do not need to read any documents in your operation set, you can
+    # execute multiple write operations as a single batch. A batch of writes
+    # completes atomically and can write to multiple documents. Batched writes
+    # are also useful for migrating large data sets to Cloud Firestore.
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # firestore.batch do |b|
+    #   # Set the data for NYC
+    #   b.set("cities/NYC", { name: "New York City" })
+    #
+    #   # Update the population for SF
+    #   b.update("cities/SF", { population: 1000000 })
+    #
+    #   # Delete LA
+    #   b.delete("cities/LA")
+    # end
+    # ```
+    #
+    # ## Deleting data
+    #
+    # Use {Firestore::DocumentReference#delete} to delete a document from Cloud
+    # Firestore:
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a document reference
+    # nyc_ref = firestore.doc "cities/NYC"
+    #
+    # nyc_ref.delete
+    # ```
+    #
+    # To delete specific fields from a document, use the
+    # {Firestore::FieldValue.delete} method when you update a document:
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a document reference
+    # nyc_ref = firestore.doc "cities/NYC"
+    #
+    # field_delete = Google::Cloud::Firestore::FieldValue.delete
```

Comment:
I would use the following for this code example:

```ruby
nyc_ref.update({ name: "New York City",
                 trash: firestore.field_delete })
```

Users can call `Google::Cloud::Firestore::FieldValue.delete` to get the value, but I think we should give higher preference to showing `Google::Cloud::Firestore::Client#field_delete` instead.

Reply:
OK, i'll change it, thanks.